src/App.jsx

```diff
@@ -14,45 +17,106 @@ export const goodsFromServer = [
-        type="button"
-        className="button is-success is-light"
-      >
-        Sort by length
-      </button>
-
-      <button
-        type="button"
-        className="button is-warning is-light"
-      >
-        Reverse
-      </button>
-
-      <button
-        type="button"
-        className="button is-danger is-light"
-      >
-        Reset
-      </button>
-    </div>
+const SORT_FIELD_NAME = 'name';
+const SORT_FIELD_LENGTH = 'length';
+
+function getPreperedGoods(goods, { sortField, reverse }) {
+  const preperedGoods = goods.map(good => (
+    {
+      name: good,
+      length: good.length,
+    }
+  ));
+
+  if (sortField) {
+    preperedGoods.sort((good1, good2) => {
+      switch (sortField) {
+        case SORT_FIELD_NAME:
+          return good1[sortField].localeCompare(good2[sortField]);
+
+        case SORT_FIELD_LENGTH:
+          return good1[sortField] - good2[sortField];
+
+        default:
+          return 0;
+      }
+    });
+  }
+
+  if (reverse) {
+    preperedGoods.reverse();
+  }
+
+  return preperedGoods;
+}
+
+export const App = () => {
+  const [sortField, setSortField] = useState('');
+  const [reverse, setReverse] = useState(false);
 
-    <ul>
-      <li data-cy="Good">Dumplings</li>
-      <li data-cy="Good">Carrot</li>
-      <li data-cy="Good">Eggs</li>
-      <li data-cy="Good">Ice cream</li>
-      <li data-cy="Good">Apple</li>
-      <li data-cy="Good">...</li>
-    </ul>
-  </div>
-);
+  const localeGoods = getPreperedGoods(
+    goodsFromServer, { sortField, reverse },
+  );
+
+  const handleClickReset = () => {
+    setSortField('');
+    setReverse(false);
+  };
+
+  const isTrue = sortField || reverse;
+
+  return (
+    <div className="section content">
+      <div className="buttons">
+        <button
+          onClick={() => setSortField(SORT_FIELD_NAME)}
+          type="button"
+          className={
+            cn('button', 'is-info', {
+              'is-light': sortField !== SORT_FIELD_NAME,
+            })
+          }
+        >
+          Sort alphabetically
+        </button>
+
+        <button
+          onClick={() => setSortField(SORT_FIELD_LENGTH)}
+          type="button"
+          className={
+            cn('button', 'is-info', {
+              'is-light': sortField !== SORT_FIELD_LENGTH,
+            })
+          }
+        >
+          Sort by length
+        </button>
+
+        <button
+          onClick={() => setReverse(!reverse)}
+          type="button"
+          className={cn('button', 'is-warning', { 'is-light': !reverse })}
+        >
+          Reverse
+        </button>
+
+        {isTrue
+          && (
+            <button
+              onClick={handleClickReset}
+              type="button"
+              className="button is-danger is-light"
+            >
+              Reset
+            </button>
+          )}
+      </div>
+
+      <ul>
+        {localeGoods.map(good => (
+          <li key={uuidv4()} data-cy="Good">{good.name}</li>
```

Comment:
❗️❗️❗️very bad idea to generate key here ❗️❗️❗️
key should be not only unique it should be stable, you set keys to avoid rerender the whole list on every action, but when you generate on every action new key the whole list will be rerendered
you will not notice loss of speed on such small list, but when it will be bigger, it will be trouble [screen](https://prnt.sc/6DQEOcntmfGC), [link](https://react.dev/learn/rendering-lists)
here better to use names of goods as keys